app/src/common/shared/com/igalia/wolvic/speech/MKSpeechRecognizer.java

```diff
@@ -30,11 +36,26 @@ public class MKSpeechRecognizer implements SpeechRecognizer, ISpeechRecognitionL
     // temporary
     private static float FINAL_CONF = 1.0f;
 
+    private static final List<String> mSupportedLanguages = Arrays.asList(
+            LocaleUtils.DEFAULT_LANGUAGE_ID,
+            "en-US", "zh-CN", "ja-JP", "fr-FR", "de-DE", "es-ES", "ru-RU", "ko-KR",
+            "it-IT", "pl-PL", "sv-SE", "fi-FI", "ar-SA", "id-ID", "th-TH", "he-IL");
+
+    private static final List<String> mSupportedLanguagesNames = new ArrayList<>();
```

Comment:
Sorry for not spotting this before but as we don't allow duplicates we should use a Set instead of a List. That would save us some headaches just in case we eventually fetch those from some service and end up with duplicates somehow.

After a second thought I realized that you have to do operations with the indexes of the Array so I guess using sets is not an option. Anyway food for thought for eventual followups.

Reply:
This is related to the problem that the speech recognizer code is too tied to the UI. We can fix it as part of https://github.com/Igalia/wolvic/issues/151